features/bootstrap/FeatureContext.php

```diff
@@ -67,6 +67,22 @@ public function alterUserParameters(EntityScope $scope) {
     }
   }
 
+    /**
+     * Hook into user creation to test `@beforeEntityCreate`
+     *
+     * @beforeEntityCreate
+     */
+    public function alterEntityParameters(EntityScope $scope)
+    {
+        // @see `features/entity.feature`
+        // Concatenate 'First name' and 'Last name' to form user name.
+        $entity = $scope->getEntity();
+        if (isset($entity->{"First name"}) && isset($entity->{"Last name"})) {
+            $entity->name = $entity->{"First name"} . ' ' . $entity->{"Last name"};
+            unset($entity->{"First name"}, $entity->{"Last name"});
+        }
```

Comment:
Yes, it would be great if these entity hooks could know what entity type they are being passed. That could be achieved by a "step_type" property, the same awful hack I'm using to keep track of bundles using "step_bundle" without colliding with entities that have fields of their own called "bundle" or "type". I agree we should create a real object to avoid these awful workarounds.

 However, this "step_bundle" syntax has already been committed on the Drupal Driver, and the code is usable without adding a "step_type" argument. I suggest we accept this PR as is, and then rearchitect to use a real class in a subsequent PR as that naturally has consequences/uses beyond this EntityContext. (Although that could be a BC break for anyone who wrote hooks or cleanup code expecting step_bundle to be there). I've opened #337 to discuss the bigger issues here.